tests/test_visitors/conftest.py

```diff
@@ -18,7 +18,7 @@ def assert_errors():
     """Helper function to assert visitor violations."""
     def factory(visitor: BaseVisitor, errors: Sequence[str]):
         for index, error in enumerate(visitor.violations):
-            assert len(errors) > index, [
+            assert index < len(errors), [
```

Comment:
It seems to me that this order was perfectly fine. Why? Because we had two variables on the both sides.

We should not count method or function calls as some kind of special case. In other words, `len(errors) > index` must be allowed.

Please, add a test case for:
1. this particular case
2. some other complex examples: like `index > index - some + random()`
3. method calls: `index > some_object.get_index()`